core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -691,6 +685,33 @@ protected void closeFailed(Throwable cause) {
         private void logStopFailure(Throwable cause) {
             logger.warn("Failed to stop a server: {}", cause.getMessage(), cause);
         }
+
+        private void shutdownExecutor(Iterable<ScheduledExecutorService> executors) {
+            for (ScheduledExecutorService executor : executors) {
+                if (executor instanceof UnstoppableScheduledExecutorService) {
+                    executor = ((UnstoppableScheduledExecutorService) executor).getExecutorService();
+                }
+
```

Comment:
Shoudln't we also do:
```java
if (executor instanceof BlockingTaskExecutor) {
    (((BlockingTaskExecutor) executor).unwrap()).shutdown();
}
```

Reply:
I think `BlockingTaskExecutor#shutdown` should shut down the wrapped executor?

Reply:
Oops, that's correct. Sorry about the noise. 😅